pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/TransactionsBase.java

```diff
@@ -432,4 +434,35 @@ protected void validateTopicName(String property, String namespace, String encod
             throw new RestException(Response.Status.PRECONDITION_FAILED, "Topic name is not valid");
         }
     }
+
+    protected void internalScaleTransactionCoordinators(int replicas, boolean authoritative) {
+        final int maxPartitions = pulsar().getConfig().getMaxNumPartitionsPerPartitionedTopic();
+        if (maxPartitions > 0 && replicas > maxPartitions) {
+            throw new RestException(Response.Status.NOT_ACCEPTABLE,
+                    "Number of partitions should be less than or equal to " + maxPartitions);
+        }
+        validateTopicOwnership(SystemTopicNames.TRANSACTION_COORDINATOR_ASSIGN, authoritative);
```

Comment:
validateTopicOwnership -> validateTopicOwnershipAsync

Reply:
This is for authentication, just like `internalUpdatePartitionedTopic` in `PersistentTopicsBase`.
It should be sync.